includes/class-wcs-action-scheduler-customer-notifications.php

```diff
@@ -0,0 +1,233 @@
+<?php
+
+/**
+ * Scheduler for subscription events that uses the Action Scheduler
+ *
+ * @class     WCS_Action_Scheduler
+ * @version   1.0.0 - Migrated from WooCommerce Subscriptions v2.0.0
+ * @package   WooCommerce Subscriptions/Classes
+ * @category  Class
+ */
+class WCS_Action_Scheduler_Customer_Notifications extends WCS_Scheduler {
+
+	protected int $hours_offset;
+
+	protected DateInterval $time_delta;
+
+	protected array $notification_actions = array(
+		'woocommerce_scheduled_subscription_customer_notification_trial_expiring',
+		'woocommerce_scheduled_subscription_customer_notification_expiration',
+		'woocommerce_scheduled_subscription_customer_notification_manual_renewal',
+		'woocommerce_scheduled_subscription_customer_notification_auto_renewal',
+	);
+
+	public function __construct() {
+		parent::__construct();
+
+		//TODO: make number of hours configurable.
+		$this->hours_offset = 3 * DAY_IN_SECONDS / HOUR_IN_SECONDS; // aka 3 days in hours.
+
+		$this->time_delta = new DateInterval( "PT{$this->hours_offset}H" );
+
+	}
+
+	public function set_date_types_to_schedule() {
+		$date_types_to_schedule = wcs_get_subscription_date_types();
+		unset( $date_types_to_schedule['start'] );
+
+		//TODO: filter?
+		$this->date_types_to_schedule = array_keys( $date_types_to_schedule );
```

Comment:
This might also concern the global toggle of this feature. Since it is initially disabled, we need a reliable method to control its access. Adding a filter here could introduce some complexity, but there is a crucial difference between this class and the `WCS_Action_Scheduler`, in that this class requires a toggle to enable or disable the feature. 🤔 

A follow-up question here would be related to how we handle this "global toggle." Should that be a checkbox? Or should we consider an empty value in the "days input" to indicate that it is disabled? I lean more toward the latter option.

Reply:
As I'm looking at it, I can see the emails in Woo have their own toggle for enabling/disabling, maybe we can reuse that (and if we provide a global disable, we can just set this toggle for all notifications?)

Reply:
Yeah, we probably don't need a filter here at this point. 

I also think that we could pass the data type into the days-getter-filter and allow developers to disable specific types (e.g., by returning `0`.) 

Related question: Should we schedule notifications for disabled emails (based on the UI toggle)? I think we still should. This is different from disabling by returning `0` from the filter above, as we cannot schedule "0 days before the event".

Reply:
> Should we schedule notifications for disabled emails (based on the UI toggle)? I think we still should.

Yes, I think we should and the code indeed does that.